vignettes/build-a-basic-package.Rmd

```diff
@@ -0,0 +1,936 @@
+if (is.null(getOption("knitr.in.progress"))) {
+  setwd(ff_path)
+}
+
+getwd()
+knitr::opts_knit$get("root.dir")
+```
+
+```{r check-wd, include = params$debug}
+## make sure new, correct wd is in effect
+## needs to be in a separate chunk, because knitr
+getwd()
+```
+
+If you use RStudio, navigate to this directory and double click on `foofactors.Rproj` to launch a new instance of RStudio in the Project that is also your foofactors package. RStudio has special handling for packages and you should now see a *Build* tab in the same pane as *Environment* and *History*.
+
+What files are in this new directory? Here's a listing (locally, you can consult your file browser):
+
+```{r init-list-files}
+cbind(listing_1 <- list.files(all.files = TRUE, no.. = TRUE))
+```
+
+  * `.gitignore` anticipates Git usage and ignores some standard, behind-the-scenes files created by R and RStudio. Even if you do not plan to use Git, this is harmless.
+  * `.Rbuildignore` lists files that we need to have around but that should not be included when building the R package from this source.
+  * `DESCRIPTION` provides [metadata about your package](http://r-pkgs.had.co.nz/description.html). We edit this shortly.
+  * `foofactors.Rproj` is the file that makes this directory an RStudio Project. Even if you don't use RStudio, this file is harmless. Or you can suppress its creation with `create(..., rstudio = FALSE)`.
+  * [`NAMESPACE`](http://r-pkgs.had.co.nz/namespace.html) declares the functions your package exports for external use and the external functions your package imports from other packages. At the moment, it holds temporary-yet-functional placeholder content.
+  * The `R/` directory is the ["business end" of your package](http://r-pkgs.had.co.nz/r.html). It will soon contain `.R` files with function definitions.
+
+## `use_git()` for version control
+
+*This is optional, but a recommended practice in the long-term. If you don't use Git, simply ignore this and subsequent instructions related to version control.*
+
+The foofactors directory is an R source package and an RStudio Project. Now we make it also a Git repository, with `devtools::use_git()`.
+
+```{r use-git}
+use_git()
+```
+
+```{r git2r-begin, include = params$debug}
+library(git2r)
+#dir.exists(ff_path)
+discover_repository(ff_path)
+```
+
+What's new? Only the creation of a `.git` directory, which is hidden in most contexts, including the RStudio file browser. Its existence is evidence that we have indeed initialized a Git repo here.
+
+```{r post-git-init-list-files}
+listing_2 <- dir(all.files = TRUE, no.. = TRUE)
+cbind(setdiff(listing_2, listing_1))
+```
+
+If you use RStudio, quit and relaunch this Project, by double clicking on `foofactors.Rproj`. Now, in addition to package development support, you have access to a basic Git client in the *Git* tab of the *Environment/History/Build* pane. Click on History (the clock icon) and you should see the initial commit made by `use_git()`:
```

Comment:
We should be able to relaunch automatically - https://github.com/r-lib/usethis/issues/42